glean-core/src/ping/mod.rs

```diff
@@ -240,8 +240,15 @@ impl PingMaker {
     ///
     /// The directory will be created inside the `data_path`.
     /// The `pings` directory (and its parents) is created if it does not exist.
-    fn get_pings_dir(&self, data_path: &Path) -> std::io::Result<PathBuf> {
-        let pings_dir = data_path.join("pending_pings");
+    fn get_pings_dir(&self, data_path: &Path, ping_type: Option<&str>) -> std::io::Result<PathBuf> {
+        // Use a special directory for deletion_request pings
+        let pings_dir = match ping_type {
+            Some(ping_type) if ping_type == "deletion_request" => {
```

Comment:
I just realized one thing: what if some library or application attempts to register a `deletion_request` ping themselves? Would Glean make it fail at build time? Would it race here? Not for fixing in this bug, but it might be worth filing a bug about this

Reply:
Same as with baseline and metrics: We don't allow it (that is: glean_parser will fail such a definition)

Reply:
Oh, right. Sorry for the noise :)